src/app/common/components/MustGatherBtn.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { WarningTriangleIcon } from '@patternfly/react-icons';
+import { MustGatherContext } from '@app/common/context';
+import { getMustGatherApiUrl } from '@app/queries/helpers';
+import { IMustGatherResponse } from '@app/client/types';
+import { UseQueryResult } from 'react-query';
+import { IPlan } from '@app/queries/types';
+
+interface IMustGatherBtn {
+  mg: IMustGatherResponse | undefined;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+  customName: string;
+  type: 'plan' | 'vm';
+}
+
+export const MustGatherBtn: React.FunctionComponent<IMustGatherBtn> = ({
+  mg,
+  mustGathers,
+  customName,
+  type,
+}) => {
+  const {
+    setMustGatherModalOpen,
+    setActiveMustGather,
+    mustGatherWatchList,
+    setMustGatherWatchList,
+  } = React.useContext(MustGatherContext);
+
+  return (
+    <>
+      {mg?.status === 'completed' && mg?.['archive-name'] ? (
+        <Tooltip
+          content={
+            !mustGathers.isSuccess
+              ? `Could not reach must gather service.`
+              : `${mg?.['archive-name']} available for download`
+          }
+        >
+          <Button
+            isAriaDisabled={!mustGathers.isSuccess}
+            variant="secondary"
+            component={'a'}
+            target="_blank"
+            download
+            href={getMustGatherApiUrl(`must-gather/${mg?.['id']}/data`)}
+          >
+            Download logs
+          </Button>
+        </Tooltip>
+      ) : (
+        <>
+          <Tooltip
+            content={
+              !mustGathers.isSuccess
+                ? `Could not reach must gather service.`
+                : mg?.status === 'inprogress'
+                ? `Collecting ${type === 'plan' ? 'migration plan' : 'VM migration'} logs.`
+                : mg?.status === 'new'
+                ? `Must gather queued for execution.`
+                : mg?.status === 'error'
+                ? `Could not complete must gather for ${mg['custom-name']}`
+                : `Collects the current ${
+                    type === 'plan' ? 'migration plan' : 'VM migration'
+                  } logs and creates a tar archive file for download.`
+            }
+          >
+            <Button
+              icon={mg?.status === 'error' ? <WarningTriangleIcon /> : null}
+              isLoading={
+                mustGatherWatchList.map((mg) => mg.name).includes(customName) &&
+                !mustGathers.isError &&
+                (mg?.status === 'inprogress' || mg?.status === 'new')
+              }
+              isAriaDisabled={
+                (mustGatherWatchList.map((mg) => mg.name).includes(customName) &&
```

Comment:
Is it worth factoring out `mustGatherWatchList.map((mg) => mg.name).includes(customName) && (mg?.status === 'inprogress' || mg?.status === 'new')` into some variable, or am I overthinking this? Not even sure what we'd call it.

Reply:
Yea, it starts to feel like the plan state/status.. it's mainly just the two scenarios for now so if you're cool with it we can leave as it, and if another condition comes into play we can step back and think about what they have in common - should help the naming of the abstraction.

Reply:
That sounds good. We'll have plenty of opportunities to abstract better later.